### PR TITLE
Prevent deadlocks on components and locale updates

### PR DESCRIPTION
**Pull request by petersg83**
### What does it do?

It uses sequential updates instead of Promise.all to avoid deadlocks

### Why is it needed?

Because otherwise it can prevent to create a new locale for an entity or to save an entry with components

### How to test it?

It's very difficult to test it as the deadlocks are random. Contact @Marc-Roig so he can show you how to :p

### Related issue(s)/PR(s)

#15202

### CODE REVIEW

**Comment by Boegie19** [comment]
users and permisisons also has one don't know if this should be in this pr or a different one?

https://github.com/strapi/strapi/blob/main/packages/plugins/users-permissions/server/services/role.js#L138-L143

the https://github.com/strapi/strapi/pull/12346 pr also would fix this

**Comment by jhoward1994** [comment]
There are also some other uses of Promise.all() in the entity-service and i18n plugin:

https://github.com/strapi/strapi/blob/62e3f9325196e0e67abc6cb893d3cdd92081ad34/packages/core/strapi/lib/services/entity-service/index.js#L253-L263

https://github.com/strapi/strapi/blob/bd201cd0541c10a7e454ca543d19305c046d5646/packages/plugins/i18n/server/migrations/field/migrate.js#L9-L23

Could these lead to deadlocks? And if not should we change the syntax anyway for consistency?

**Review by nathan-pichon — APPROVED** [review]
LGTM as a quick fix.
But I'm persuaded that we should find a way to do ops in parallel as this can decrease performances over huge sets of data. Also, I think we need to mention this point in the release patch note.
Maybe we can put this element to be addressed in the v5 architecture spec process.

**Comment by petersg83** [comment]
Thank you for your reviews!

Yes, we let some `Promise.all` here and there in the code as we think they are not causing deadlocks. We prefer to wait for users to tell us if they still see deadlocks rather than slowing everything.

We just did another [PR](https://github.com/strapi/strapi/pull/15554) on the same part of the code to fix performance issues and it appears... it may also fix the deadlocks! So we would like to first merge https://github.com/strapi/strapi/pull/15554 and see if it fixes the deadlocks issues for users before merging this PR :)
In the meantime this PR is on hold then

**Pull request merged**
